packages/jest-cli/src/cli/args.js

```diff
@@ -85,8 +85,8 @@ export const options = {
   bail: {
     alias: 'b',
     default: undefined,
-    description: 'Exit the test suite immediately upon the first failing test.',
-    type: 'boolean',
```

Comment:
Should the type stay as number?

Reply:
No, if you enforce `number` then yargs won't handle invalid data.  It would make it so it's not backwards compatible (where both nothing and `true/false` are accepted)